### PR TITLE
ref(routes): Simplify routing tree using custom route builder

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This helps reduce the current complexity of reading the route tree by introducing some logic which builds the two org and orgless routes from a single route.

This only changes the simplest cases. There are a number of more complicated routes that can also be simplified that I will do in a follow up.

### CODE REVIEW

File: static/app/routes.tsx [hunk 1]
```diff
@@ -1521,49 +1564,19 @@ function buildRoutes() {
...
-        path="/organizations/:orgId/release-thresholds/"
-        component={withDomainRedirect(NoOp)}
-        key="org-release-thresholds"
-      >
-        <IndexRoute
-          component={make(() => import('sentry/views/releases/thresholdsList'))}
-        />
-      </Route>
-    </Fragment>
+    <Route path="/release-thresholds/" withLegacyPath>
```
Comment (evanpurkhiser) [inline static/app/routes.tsx:1567]: In this case the `NoOp` component is not required and will be added as needed.

File: static/app/routes.tsx [hunk 2]
```diff
@@ -133,12 +201,9 @@ function buildRoutes() {
...
   //
-  //   When adding new routes make sure you have both a route that starts
-  //   with `/organizations/:orgId` and also 'customer-domains' route that
-  //   does not include `/organizations/:orgId`. Often you'll only need to
-  //   worry about this for the container route for that section of the UI.
-  //   Child routes should access the current organization with `useOrganization()`
-  //   or `withOrganization()` methods.
+  //   When adding new top-level organization routes, be sure the top level
+  //   route includes withOrgPath to support installs that are not using
+  //   customer domains.
```
Comment (gggritso) [inline static/app/routes.tsx:206]: If this is really important (which it is) might be worth making a component called `TopLevelRoute` and another called `Route` so this error is harder to make

File: static/app/routes.tsx [hunk 3]
```diff
@@ -30,16 +30,84 @@ import RouteNotFound from 'sentry/views/routeNotFound';
...
+// We override the createRouteFromReactElement property to provide support for
+// the withOrgPath property.
+//
+// XXX(epurkhiser): It is important to note! The `Route` component is a
+// CONFIGURATION ONLY COMPONENT. It DOES NOT render! This function is part of
+// the react-router magic internals that are used to build the route tree by
+// traversing the component tree, that is why this logic lives here and not
+// inside a custom Route component.
+//
+// To udnerstand deepr how thsi works, see [0].
```
Comment (gggritso) [inline static/app/routes.tsx:74]:
```suggestion
// To understand deeper how this works, see [0].
```

**Review by gggritso — APPROVED** [review]
🤸🏻

File: static/app/routes.tsx (hunk 2 above)
Comment (evanpurkhiser) [reply static/app/routes.tsx:206]: that's not a bad idea 👍

**Pull request merged**